### DISCUSSION TITLE
Converting class property writing to `camelCase` <--> `snake_case`

### DISCUSSION THREAD

**Discussion by andrey-helldar** [PHP-CS-Fixer/PHP-CS-Fixer#8206] [Feature ideas]
## Rule request

I tried to find this rule by words like `property`, `case` and `casing`, but couldn't.

Is there a rule for transforming the appearance of variables?

### Code snippet that shows how the rule would fix PHP code

```diff
<?php

class Foo
{
-    protected int $some_key = 1;
+    protected int $someKey = 1; 

    public function getSome(): int
    {
-        return $this->some_key;
+        return $this->someKey; 
    }
}
```

And also in reverse order from `camelCase` to `snake_case`.

**Comment by scasei** [comment]
Here is a propsal: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/5097

---

FYI: Mentioned PR is for variables, not for properties.

**Comment by Wirone** [comment]
I believe we need better specification for this. Which visibility levels should be processed? Please keep in mind that touching public/protected properties this way is a BC break and Fixer does not have inheritance context during analysis. I am pretty sure this should be risky fixer.

**Comment by andrey-helldar** [comment]
This should definitely be a risky fixer because it changes the format of keys and the larger the size of the project, the higher the probability that it may miss some keys or convert something it should not. For example, dock blocks for "magic" fields of some classes.

As for visibility, I would add a "visibility" option with the ability to select values: "all", "public", "protected", "private". So that the developer could choose which fields should be processed.
